src/main.rs

```diff
@@ -143,13 +143,20 @@ pub fn main() {
             let args = std::env::args().skip(2);
             if let Some(first) = target.kind.get(0) {
                 if target.kind.len() > 1 || first.ends_with("lib") {
+                    println!("compiling library");
```

Comment:
it's rather hard seeing what's going on. Especially since some errors apparently don't have spans so there's no file or other information. (I tried it out with an example that had no  `main` function)